### PR TITLE
Update gem versions in modules

### PR DESCRIPTION
**Pull request by ekohl**
This should bring CI support to test Puppet 8.

* [ ] https://github.com/theforeman/puppet-candlepin/pull/241
* [ ] https://github.com/theforeman/puppet-certs/pull/427
* [ ] https://github.com/theforeman/puppet-dhcp/pull/225
* [ ] https://github.com/theforeman/puppet-dns/pull/240
* [ ] https://github.com/theforeman/puppet-foreman/pull/1133
* [ ] https://github.com/theforeman/puppet-foreman_proxy/pull/814
* [ ] https://github.com/theforeman/puppet-foreman_proxy_content/pull/461
* [ ] https://github.com/theforeman/puppet-foreman_scap_client/pull/75
* [ ] https://github.com/theforeman/puppet-katello/pull/476
* [ ] https://github.com/theforeman/puppet-katello_devel/pull/294
* [ ] https://github.com/theforeman/puppet-katello_devel/pull/297
* [ ] https://github.com/theforeman/puppet-motd/pull/15
* [ ] https://github.com/theforeman/puppet-pulpcore/pull/309
* [ ] https://github.com/theforeman/puppet-puppet/pull/888
* [ ] https://github.com/theforeman/puppet-puppetserver_foreman/pull/36
* [ ] https://github.com/theforeman/puppet-tftp/pull/143

### CODE REVIEW

**Comment by ekohl** [comment]
Still needs https://github.com/voxpupuli/puppet-lint-spaceship_operator_without_tag-check/pull/18 now. Will send a batch of PRs when it's released.

**Comment by ekohl** [comment]
Somehow bundler can't find version 1.0.0 so I'm still stumped.

File: moduleroot/Gemfile.erb [hunk 1]
```diff
@@ -3,14 +3,11 @@
...
-gem 'puppet', ENV['PUPPET_GEM_VERSION'] || '>= 5.5', groups: ['development', 'test']
+gem 'puppet', ENV.fetch('PUPPET_GEM_VERSION', '>= 7'), groups: ['development', 'test']
 gem 'rake'
 
 <% (@configs['required'] + (@configs['extra'] || [])).each do |gem| -%>
 gem '<%= gem['gem'] %>'<%= ", '#{gem['version']}'" if gem['version'] %><%= ", #{gem['options'].inspect}" if gem['options'] %><%= " if RUBY_VERSION #{gem['ruby_version']}" if (gem['ruby_version']) %>
 <% end -%>
 
-# Pin rdoc to prevent updating bundled psych (https://github.com/ruby/rdoc/commit/ebe185c8775b2afe844eb3da6fa78adaa79e29a4)
-gem 'rdoc', '< 6.4'
```
Comment (ekohl) [inline moduleroot/Gemfile.erb:14]: Should we keep this on Ruby 2 in to prevent updating psych on Ruby 2.7?

**Comment by ekohl** [comment]
I think this is ready for review now. All PRs have been updated and I expect them to pass now.

**Comment by ekohl** [comment]
Still required to fix:
* https://github.com/theforeman/puppet-puppet/pull/888#issuecomment-1751084397
* https://github.com/theforeman/puppet-katello/pull/476#issuecomment-1761722586 (likely an unrelated failure, but I want it to be green)

File: moduleroot/.github/workflows/ci.yml.erb [hunk 2]
```diff
@@ -3,8 +3,6 @@ name: CI
 
 on:
   pull_request:
-  schedule:
-    - cron: '4 4 * * *'
```
[the remark this answers is not in the document]
Comment (ekohl) [reply moduleroot/.github/workflows/ci.yml.erb:7]: I also debated that. In practice I rarely look at it and @ehelms wasn't even aware that we had it. So for now I'm leaning to keeping it disabled.

[the remark this answers is not in the document]
Comment (evgeni) [reply moduleroot/.github/workflows/ci.yml.erb:7]: k, wfm

(I did use it the other day when for some reason the `centos:stream8` container stopped booting out of a sudden to double check that it *really* worked a few hours ago, but then again this was probably the only use of it by me :D)

[the remark this answers is not in the document]
Comment (ekohl) [reply moduleroot/.github/workflows/ci.yml.erb:7]: I also used the other day to see that puppet-katello was already failing before I got started, but I wonder if it still makes sense. At least with the reusable workflows we'll have less duplication if we have 2 separate files.

[the remark this answers is not in the document]
Comment (evgeni) [reply moduleroot/.github/workflows/ci.yml.erb:7]: technically, you could even re-use a re-usable workflow: `ci.yml.erb` becomes `workflow.yml.erb` with `on: workflow_call` and then we have `ci.yml` and `cron.yml` that both call that…

or you use ERB to generate two "identical" files with different `on` statements.

or I shut up and we get it in as is and think later how to overengineer this ;)

[the remark this answers is not in the document]
Comment (ekohl) [reply moduleroot/.github/workflows/ci.yml.erb:7]: You may be able to trick modulesync with using a symlink and read `__FILE__` in the `on` condition. Does that qualify as overengineering? ;)

[the remark this answers is not in the document]
Comment (evgeni) [reply moduleroot/.github/workflows/ci.yml.erb:7]: It's a "smart" solution. "smart" solutions tend to explode in spectacular ways down the road. ;-)

File: moduleroot/Gemfile.erb (hunk 1 above)
Comment (ekohl) [reply moduleroot/Gemfile.erb:14]: I'm going to merge it without it and then see

**Pull request merged**